R/as_polars_df.R

```diff
@@ -74,7 +74,15 @@ as_polars_df <- function(x, ...) {
 #' @rdname as_polars_df
 #' @export
 as_polars_df.default <- function(x, ...) {
-  dtype <- infer_polars_dtype(x, ...)
+  dtype <- try_fetch(
+    infer_polars_dtype(x, ...),
+    error = function(cnd) {
+      abort(
+        "This object is not supported for the default method of `as_polars_df()` because it can't be converted to a polars Series.",
```

Comment:
This could be made shorter IMO, e.g:

```suggestion
       paste0("Cannot convert ", obj_type_friendly(x), " to a Polars Series, and hence to a Polars DataFrame.")
```
or sth similar.